main/maven/src/mill/main/maven/Plugins.scala

```diff
@@ -0,0 +1,126 @@
+          dom / "source" <> (options += "-source" += _)
+          dom / "target" <> (options += "-target" += _)
+        } else {
+          options += "--release" += release.getValue
+        }
+        dom / "encoding" <> (options += "-encoding" += _)
+        dom / "compilerArgs" <*> (options += _)
+      }
+
+      options.result()
+    }
+  }
+
+  /**
+   * @see [[https://www.mojohaus.org/build-helper-maven-plugin/index.html]]
+   */
+  object `build-helper-maven-plugin` {
+
+    type Sources = Seq[String]
+    type Resources = Seq[String]
+    type CompileSources = (Sources, Resources)
+    type TestSources = (Sources, Resources)
+
+    def find(model: Model): Option[Plugin] =
+      Plugins.find(model, "org.codehaus.mojo", "build-helper-maven-plugin")
+
+    def added(model: Model): (CompileSources, TestSources) = {
+      val sources = Seq.newBuilder[String]
+      val resources = Seq.newBuilder[String]
+      val testSources = Seq.newBuilder[String]
+      val testResources = Seq.newBuilder[String]
+      find(model).foreach { plugin =>
+        val dir = os.Path(model.getProjectDirectory)
+        def ref(path: String): String = {
+          val rel = os.Path(path).relativeTo(dir)
+          s"PathRef(millSourcePath / \"$rel\")"
+        }
+
+        plugin.getExecutions.iterator().asScala.foreach { execution =>
+          dom(execution).foreach { dom =>
+            execution.getGoals.iterator().asScala.foreach {
+              case "add-source" =>
+                dom / "sources" * "source" <> (path => sources += ref(path))
+              case "add-resource" =>
+                dom / "resources" * "resource" / "directory" <> (path => resources += ref(path))
+              case "add-test-source" =>
+                dom / "sources" * "source" <> (path => testSources += ref(path))
+              case "add-test-resource" =>
+                dom / "resources" * "resource" / "directory" <> (path =>
+                  testResources += ref(path)
+                )
+              case _ =>
+            }
+          }
+        }
+      }
+      ((sources.result(), resources.result()), (testResources.result(), testResources.result()))
+    }
+  }
+
+  private implicit class NullableDomOps(val self: Xpp3Dom) extends AnyVal {
+
+    def /(name: String): Xpp3Dom =
+      if (null == self) self else self.getChild(name)
+
+    def *(name: String): Iterator[Xpp3Dom] =
+      if (null == self) Iterator.empty[Xpp3Dom] else self.getChildren(name).iterator
+
+    def <>(f: String => Unit): Unit =
+      if (null != self) f(self.getValue)
+
+    def <*>(f: String => Unit): Unit =
+      if (null != self) self.getChildren.iterator.foreach(dom => f(dom.getValue))
+  }
+
+  private implicit class IteratorDomOps(val self: Iterator[Xpp3Dom]) extends AnyVal {
+
+    def /(name: String): Iterator[Xpp3Dom] =
+      self.map(_.getChild(name))
+
+    def <>(f: String => Unit): Unit =
+      if (null != self) self.foreach(dom => f(dom.getValue))
+  }
```

Comment:
Let's give these non-operator names. How about something like this

```scala
def child
def children
def foreachValue
def foreachChildValue
```

In general we should limit our use of operators to things that will be both used extremely densely (i.e. more than one method call per line) and extremely commonly (e.g. all over a big part of the codebase). I understand what you are trying to do here but I don't think this quite makes the cut, so we should fall back to normal extension methods even if it's a bit more verbose

Reply:
Renamed extensions. Please review.